### PR TITLE
Add support for checks in manifest2proto conversion.

### PR DESCRIPTION
**Pull request by bgogul**

### CODE REVIEW

File: src/tools/tests/manifest2proto-test.ts [hunk 1]
```diff
@@ -588,6 +588,171 @@ describe('manifest2proto', () => {
...
+
+  it('encodes particle spec with compound checks (>2 children)', async () => {
+    const manifest = await Manifest.parse(`
+      particle Test in 'a/b/c.js'
+        private: reads {name: Text}
+        public: reads {name: Text}
+        check private is private_tag and is secret and is something
+        check public is public_tag or (is secret and is something)
+     `);
+    //        check public is public_tag or is something
```
Comment (csilvestrini) [inline src/tools/tests/manifest2proto-test.ts:693]: Delete this??

Comment (bgogul) [reply src/tools/tests/manifest2proto-test.ts]: Oops, done!

**Pull request merged**